aiohttp/connector.py

```diff
@@ -1107,6 +1107,7 @@ def drop_exception(fut: "asyncio.Future[List[Dict[str, Any]]]") -> None:
                     fut.result()
 
             host_resolved.add_done_callback(drop_exception)
+            host_resolved.cancel()
```

Comment:
Looking at this again, the comment above very clearly says we shouldn't cancel this task, which is why it's shielded. Without a test, I don't think this will be going anywhere.